packages/edit-site/src/components/global-styles/hooks.js

```diff
@@ -0,0 +1,210 @@
+/**
+ * External dependencies
+ */
+import { get, cloneDeep, set, isEqual, has } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo, useCallback } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { useEntityProp } from '@wordpress/core-data';
+import {
+	getBlockType,
+	__EXPERIMENTAL_PATHS_WITH_MERGE as PATHS_WITH_MERGE,
+	__EXPERIMENTAL_STYLE_PROPERTY as STYLE_PROPERTY,
+} from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+
+const EMPTY_CONFIG = { isGlobalStylesUserThemeJSON: true, version: 1 };
+
+function useGlobalStylesUserConfig() {
+	const globalStylesId = useSelect( ( select ) => {
+		return select( editSiteStore ).getSettings()
+			.__experimentalGlobalStylesUserEntityId;
```

Comment:
Right now this is how we retrieve the current global styles entity ID. This is not great, we shouldn't be depending on the a setting of editSite store, The proposed endpoint here would solve this https://github.com/WordPress/gutenberg/discussions/35141